pvlib/iotools/ecmwf_macc.py

```diff
@@ -0,0 +1,313 @@
+        except TypeError:
+            params = PARAMS.get(params)
+    startdate = startdate.strftime('%Y-%m-%d')
+    stopdate = stopdate.strftime('%Y-%m-%d')
+    if not server:
+        server = ECMWFDataServer()
+    t = threading.Thread(target=target, daemon=True,
+                         args=(server, startdate, stopdate, params, filename))
+    t.start()
+    return t
+
+
+class ECMWF_MACC(object):
+    """container for ECMWF MACC reanalysis data"""
+
+    TCWV = 'tcwv'  # total column water vapor in kg/m^2 at (1-atm,25-degC)
+
+    def __init__(self, filename):
+        self.data = netCDF4.Dataset(filename)
+        # data variables and dimensions
+        variables = set(self.data.variables.keys())
+        dimensions = set(self.data.dimensions.keys())
+        self.keys = tuple(variables - dimensions)
+        # size of lat/lon dimensions
+        self.lat_size = self.data.dimensions['latitude'].size
+        self.lon_size = self.data.dimensions['longitude'].size
+        # spatial resolution in degrees
+        self.delta_lat = -180.0 / (self.lat_size - 1)  # from north to south
+        self.delta_lon = 360.0 / self.lon_size  # from west to east
+        # time resolution in hours
+        self.time_size = self.data.dimensions['time'].size
+        self.start_time = self.data['time'][0]
+        self.stop_time = self.data['time'][-1]
+        self.time_range = self.stop_time - self.start_time
+        self.delta_time = self.time_range / (self.time_size - 1)
+
+    def get_nearest_indices(self, latitude, longitude):
+        """
+        Get nearest indices to (lat, lon).
+
+        Parmaeters
+        ----------
+        latitude : float
+            Latitude in degrees
+        longitude : float
+            Longitude in degrees
+
+        Returns
+        -------
+        idx_lat : int
+            index of nearest latitude
+        idx_lon : int
+            index of nearest longitude
+        """
+        # index of nearest latitude
+        idx_lat = int(round((latitude - 90.0) / self.delta_lat))
+        # avoid out of bounds latitudes
+        if idx_lat < 0:
+            idx_lat = 0  # if latitude == 90, north pole
+        elif idx_lat > self.lat_size:
+            idx_lat = self.lat_size  # if latitude == -90, south pole
+        # adjust longitude from -180/180 to 0/360
+        longitude = longitude % 360.0
+        # index of nearest longitude
+        idx_lon = int(round(longitude / self.delta_lon)) % self.lon_size
+        return idx_lat, idx_lon
+
+    def interp_data(self, lat, lon, utc_time, key):
+        """
+        Interpolate data for UTC time using nearest indices to (lat, lon).
+
+        Parmaeters
+        ----------
+        lat : float
+            Latitude in degrees
+        lon : float
+            Longitude in degrees
+        utc_time : datetime.datetime or datetime.date
+            Naive or UTC date or datetime to interpolate
+        key : str
```

Comment:
maybe `param` would be more consistent with the other functions

Reply:
Agree